src/status_im/contexts/settings/wallet/saved_addresses/view.cljs

```diff
@@ -50,15 +60,34 @@
       :on-press        on-press-saved-address}]))
 
 (defn- header
-  [{:keys [title]}]
+  [{:keys [title index]}]
   [quo/divider-label
-   {:tight? true
-    :blur?  true}
+   {:tight?          true
+    :blur?           true
+    :container-style (when (pos? index) {:margin-top 8})}
    title])
 
-(defn- footer
+(defn- filtered-list
+  [{:keys [search-text]}]
+  [rn/flat-list
+   {:key-fn                          :address
+    :data                            (rf/sub [:wallet/filtered-saved-addresses search-text])
+    :render-fn                       saved-address
+    :shows-vertical-scroll-indicator false
+    :content-container-style         {:flex-grow 1}
+    :empty-component                 [empty-result]}])
+
+(defn- unfiltered-list
   []
-  [rn/view {:height 8}])
+  [rn/section-list
+   {:key-fn                          :title
+    :shows-vertical-scroll-indicator false
+    :sticky-section-headers-enabled  false
+    :render-section-header-fn        header
+    :sections                        (rf/sub [:wallet/grouped-saved-addresses])
```

Comment:
should it be called every time input is empty? or once at mount is enough?

Reply:
At mount is enough. I need to move this to the main view. Fundamentally, subs are cached but it's best we move to the main view.